docs/architecture/architecture.md

```diff
@@ -0,0 +1,378 @@
+[default Docker runtime (runc)](https://github.com/opencontainers/runc) with `cc-runtime`.
+
+![Runtime and virtcontainers](runtime-vc-relationship.png)
+![Docker and Clear Containers](docker-cc.png)
+
+`cc-runtime` creates a QEMU/KVM virtual machine for each container the Docker engine creates.
+
+The container process is then spawned by an [agent](https://github.com/clearcontainers/agent),
+running as a daemon on the guest operating system.
+Hyperstart opens 2 virtio serial interfaces (Control and I/O) on the guest, and QEMU exposes them
+as serial devices on the host. `cc-runtime` uses the control device for sending container
+management commands to the agent while the I/O serial device is used to pass I/O streams (`stdout`,
+`stderr`, `stdin`) between the guest and the Docker Engine.
+
+For any given container, both the init process and all potentially executed commands within that
+container, together with their related I/O streams, need to go through 2 virtio serial interfaces
+exported by QEMU. The [Clear Containers proxy (`cc-proxy`)](https://github.com/clearcontainers/proxy)
+multiplexes and demultiplexes those commands and streams for all container virtual machines.
+There is only one `cc-proxy` instance running per Clear Containers host.
+
+On the host, each container process is reaped by a Docker specific (`containerd-shim`) monitoring
+daemon. As Clear Containers processes run inside their own virtual machines, `containerd-shim`
+can not monitor, control or reap them. `cc-runtime` fixes that issue by creating an
+[additional shim process (`cc-shim`)](https://github.com/clearcontainers/shim)
+between `containerd-shim` and `cc-proxy`. A `cc-shim` instance will both forward signals and `stdin`
+streams to the container process on the guest and pass the container `stdout` and `stderr` streams
+back to the Docker engine via `containerd-shim`.
+`cc-runtime` creates a `cc-shim` daemon for each Docker container and for each command Docker
+wants to run within an already running container (`docker exec`).
+
+The container workload, i.e. the actual OCI bundle rootfs, is exported from the host to
+the virtual machine via a 9pfs virtio mount point. Hyperstart uses this mount point as the root
+filesystem for the container processes.
+
+![Overall architecture](overall-architecture.png)
+
+#### Hypervisor
+
+Clear Containers use [KVM](http://www.linux-kvm.org/page/Main_Page)/[QEMU](http://www.qemu-project.org/) to
+create virtual machines where Docker containers will run:
+
+![QEMU/KVM](qemu.png)
+
+** fixme - discuss different QEMUs - lite, q35, pc etc. **
+Although Clear Containers can run with any recent QEMU release, containers boot time and memory
+footprint are significantly optimized by using a specific QEMU version called [`qemu-lite`](https://github.com/clearcontainers/qemu/tree/qemu-lite-v2.9.0).
+
+`qemu-lite` improvements comes through a new `pc-lite` machine type, mostly by:
```

Comment:
We are using PC machine type for 3.0, I think for release perhaps it makes sense to just remove all of the permutations and just describe what we actually have?

Reply:
took a pass at this @ https://github.com/clearcontainers/runtime/wiki/Clear-Containers-Architecture#hypervisor - not sure how much detail we should go into features we aren't actually using though....